### PR TITLE
fix/memorize-game-logic

### PR DESCRIPTION
**Pull request by mosayed01**

### CODE REVIEW

**Review by Jina-n11 — APPROVED** [review]

File: feature/trvia/data/repository/src/main/java/com/chocolatecake/repository/TriviaRepositoryIml.kt [hunk 1]
```diff
@@ -122,22 +122,16 @@ class TriviaRepositoryIml @Inject constructor(
 
     override suspend fun getBoard(level: Int): BoardEntity {
         val size = when (level) {
-            1 -> 9
-            2 -> 12
+            1 -> 12
+            2 -> 16
```
Comment (Saif-Alhaider) [inline feature/trvia/data/repository/src/main/java/com/chocolatecake/repository/TriviaRepositoryIml.kt:126]: don't use hard coded values add them in a companion object

File: feature/trvia/data/repository/src/main/java/com/chocolatecake/repository/TriviaRepositoryIml.kt [hunk 2]
```diff
@@ -122,22 +122,16 @@ class TriviaRepositoryIml @Inject constructor(
...
             else -> 20
         }
```
Comment (Saif-Alhaider) [inline feature/trvia/data/repository/src/main/java/com/chocolatecake/repository/TriviaRepositoryIml.kt:128]: if your using only half size in this function then write it like this
```
val sizeInHalf = when (2) {
    1 -> 12
    2 -> 16
    else -> 20
}.div(2)
```

File: feature/trvia/data/repository/src/main/java/com/chocolatecake/repository/TriviaRepositoryIml.kt [hunk 3]
```diff
@@ -122,22 +122,16 @@ class TriviaRepositoryIml @Inject constructor(
 
     override suspend fun getBoard(level: Int): BoardEntity {
         val size = when (level) {
```
Comment (Saif-Alhaider) [inline feature/trvia/data/repository/src/main/java/com/chocolatecake/repository/TriviaRepositoryIml.kt:124]: size of what ?
give it a clear name

File: feature/trvia/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/memorize_game/MemorizeGameUIState.kt [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
     val countDownTimer: Int = 30,
     val level: Int = 1,
     val board: List<ItemGameImageUiState> = emptyList(),
     val initialBoard: List<ItemGameImageUiState> = emptyList(),
+    val selectedUserPositions: List<Int> = emptyList(),
     val boardSize: Int = 12,
     val points: Int = 0,
-    val CorrectPairPositions: Pair<Int,Int> = Pair(0,1),
     val firstUserPosition: Int? = null,
     val secondUserPosition: Int? = null,
```
Comment (Saif-Alhaider) [inline feature/trvia/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/memorize_game/MemorizeGameUIState.kt:13]: if possible don't give the ui state a null values

File: feature/trvia/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/memorize_game/MemorizeGameViewModel.kt [hunk 5]
```diff
@@ -60,19 +59,26 @@ class MemorizeGameViewModel @Inject constructor(
 
     private fun onSuccessUser(user: UserEntity) {
         _state.update {
+            val maxTime = when (user.memorizeGameLevel) {
+                1 -> 30
+                2 -> 60
+                else -> 90
```
Comment (Saif-Alhaider) [inline feature/trvia/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/memorize_game/MemorizeGameViewModel.kt:65]: don't use hard coded values

File: feature/trvia/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/memorize_game/MemorizeGameViewModel.kt [hunk 6]
```diff
@@ -92,21 +98,30 @@ class MemorizeGameViewModel @Inject constructor(
...
-            _state.update { it.copy(board = modifyBoard) }
-            delay(300)
+            toggleGameItem(itemGameImageUiState)
+            delay(500)
             saveUserPosition(itemGameImageUiState.position)
+            checkIfWinner()
         }
     }
 
+    private fun checkIfWinner() {
```
Comment (Saif-Alhaider) [inline feature/trvia/presentation/viewmodel/src/main/java/com/chocolatecake/viewmodel/memorize_game/MemorizeGameViewModel.kt:112]: your not checking if the user is winning or not your handling what will happen if the user won 
a function that starts with check means that the function must return a boolean and that's not what the function does
you could change the function name to "determineWinningOutcome" or something else

**Pull request merged**